### ISSUE TITLE
RedisSessionExpirationPolicy onDelete uses wrong key

### ISSUE THREAD

**Issue by swurzinger** [spring-projects/spring-session#1799]
The current implementation of `RedisSessionExpirationPolicy.onDelete` method uses the wrong session key for deletion, thus it doesn't delete the entry from the set.

While the `spring:session:expirations:{timestamp}` set actually contains the session id prefixed by `expires:`, the method tries to delete just the sessionId string (which is not contained in the set).

See the screenshots below for an example:

![image](https://user-images.githubusercontent.com/6787397/110824063-fd36df80-8292-11eb-8e83-e599c200c13d.png)

![image](https://user-images.githubusercontent.com/6787397/110824466-61f23a00-8293-11eb-9511-844d1e46d13d.png)

**To Reproduce**
call `deleteById` to delete a session from the repository when using the (default) RedisIndexedSessionRepository implementation. This will internall call `RedisSessionExpirationPolicy.onDelete`.

see sample below.

**Expected behavior**

The entry of the set should be deleted.

**Sample**

```
private val redisSessionRepository: RedisIndexedSessionRepository

fun deleteAllSessionsForPrincipal(principal: String) {
		val sessions: Map<String, Session> = redisSessionRepository.findByPrincipalName(principal)
		sessions.forEach { (sessionId, session) ->
				redisSessionRepository.deleteById(sessionId)
		}
}
```

**Comment by swurzinger** [comment]
I prepared a suggested fix in #1800 

this ticket seems to describe the same issue, but was closed due to insufficient description: #1515

**Comment by eleftherias** [comment]
Thanks for the report @swurzinger.
This is actually a duplicate of gh-585 as well.
I will close this as a duplicate and we can continue the conversation on that issue.
